File: n067-add-binary.js
var addBinary = function(a, b) {
    var alen = a.length
    var blen = b.length
    var carry = 0

    var ret = ""
    for (; alen > 0 || blen > 0; ) {
        abit = 0
        bbit = 0

        if (alen > 0) {
            abit = a[alen-1] - 0
            alen--
        }

        if (blen > 0) {
            bbit = b[blen-1] - 0
            blen--
        }

        console.log(abit, bbit, carry)
        sum = abit + bbit + carry

        carry = sum >= 2 ? 1 : 0
        r = sum%2

        ret = r + ret
    }

    if (carry === 1) {
        return "1" + ret
    } else {
        return ret
    }
}

File: n067-add-binary.go
func addBinary(a string, b string) string {
	aLen := len(a)
	bLen := len(b)
	// 进位
	carry := 0
	var ret string

	for aLen > 0 || bLen > 0 {
		aBit := 0
		bBit := 0

		if aLen > 0 {
			aBit = int(a[aLen-1] - '0')
			aLen--
		}

		if bLen > 0 {
			bBit = int(b[bLen-1] - '0')
			bLen--
		}

		sum := aBit + bBit + carry
		carry = sum / 2
		r := strconv.Itoa(sum % 2)
		ret = r + ret

	}

	if carry == 1 {
		return "1" + ret
	}

	return ret
}

